File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchaseStateAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.PurchaseState;

@SuppressWarnings({"unused"})
final class PurchaseStateAPI {
    static void check(final PurchaseState state) {
        switch (state) {
            case UNSPECIFIED_STATE:
            case PURCHASED:
            case PENDING:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchaseStateAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.PurchaseState

@Suppress("unused", "UNUSED_VARIABLE")
private class PurchaseStateAPI {
    fun check(state: PurchaseState) {
        when (state) {
            PurchaseState.UNSPECIFIED_STATE,
            PurchaseState.PURCHASED,
            PurchaseState.PENDING,
            -> {}
        }.exhaustive
    }
}
